tests/test_execution.py

```diff
@@ -203,8 +203,37 @@ def test_traintuple_build_failure(dockerfile, factory, client, default_dataset):
 
 
 @pytest.mark.slow
-def test_traintuple_execution_failure(factory, network, default_dataset_1):
-    """Invalid algo script is causing traintuple failure."""
+@pytest.mark.subprocess_skip
+def test_algo_build_failure(factory, network, default_dataset_1):
```

Comment:
Should this test be merged or supersed [`test_traintuple_build_failure`](https://github.com/Substra/substra-tests/blob/main/tests/test_execution.py#L191)?

Reply:
I don't think it's a good idea since it would increase complexity for no big benefit (we would have to introduce if's in the test body). Also, the second test is skipped in subprocess mode and the first test isn't.

Reply:
Parametrization aside, they seem very similar though. It seems that your new test could supersed [test_traintuple_build_failure](https://github.com/Substra/substra-tests/blob/main/tests/test_execution.py#L191).

Reply:
ah good point about that other test, I'll merge them

Reply:
No worries :)

Reply:
Thanks for the feedback, I deleted the old test - I agree it's superseded by the new one. https://github.com/Substra/substra-tests/pull/218/commits/abb20cd7ca1bc749a5f80e8452920bc8baa0ddb3